sigstore/_internal/rekor/checkpoint.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2023 The Sigstore Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Rekor Checkpoint machinery.
+"""
+
+from __future__ import annotations
+
+import base64
+import re
+import struct
+from dataclasses import dataclass
+from typing import List
+
+from pydantic import BaseModel, Field, StrictStr
+
+from sigstore._internal.keyring import KeyringSignatureError
+from sigstore._internal.rekor.client import RekorClient
+from sigstore._utils import KeyID
+
+
+@dataclass(frozen=True)
+class RekorSignature:
+    """
+    Represents a `RekorSignature` containing:
+    - the name of the signature, e.g. "rekor.sigstage.dev"
+    - the signature hash
+    - the base64 signature
+    """
+
+    # FIXME(jl): this does not feel like a de novo definition...
+    # does this exist already in sigstore-python (or its depenencices)?
```

Comment:
I don't believe this exists anywhere, so this comment is safe to remove.

OTOH, we might want to rename this to `RekorCheckpointSignature` or similar, since there are other signatures that happen in the context of Rekor.